### PR TITLE
Added domaintools byoi

### PR DESCRIPTION
**Pull request by shaniacht1**
fixes #8069
To be merged only afterDomainTools is removed from server

### CODE REVIEW

File: Integrations/integration-DomainTools.yml [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+  name: username
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: API Key
+  name: key
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: Open authentication (less secure)
```
Comment (Spivaka) [inline Integrations/integration-DomainTools.yml]: proxy?

File: Integrations/integration-DomainTools.yml [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+                Method: 'GET',
+                Headers: {
+                    Accept: ['application/json']
+                }
+            }
+        );
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+        return JSON.parse(res.Body);
```
Comment (Spivaka) [inline Integrations/integration-DomainTools.yml]: try { return JSON.parse(res.Body); }; throw 'Failed to parse JSON';
here, so if the response is not json we will get a clear error

File: Integrations/integration-DomainTools.yml [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+    };
+
+    var addKeyToJson = function(cur, toAdd){
```
Comment (Spivaka) [inline Integrations/integration-DomainTools.yml]: this is not add key to list?

File: Integrations/integration-DomainTools.yml [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+            context = {'Domain': {'Name': res.response.record_source, 'Whois': resMap}};
+        }else{
+            context = {'Domain': {'Name': res.response.record_source, 'Whois': changeKeys(toCamelCase, res.response.parsed_whois)}};
+        }
+
+        return {
+                Type: entryTypes.note,
+                Contents: res,
+                ContentsFormat: formats.json,
+                HumanReadable: md,
```
Comment (Spivaka) [inline Integrations/integration-DomainTools.yml]: human readable format markdown is missing

File: Integrations/integration-DomainTools.yml [hunk 5]
```diff
@@ -0,0 +1,541 @@
...
+                EntryContext: context
+            };
+    };
+
+    var callProfile= function(url, domain){
+        var domRes = sendRequest(url + domain + '/'+encodeToURLQuery({'api_username':params.username,'api_key':params.key}));
+        return {
+                Type: entryTypes.note,
```
Comment (Spivaka) [inline Integrations/integration-DomainTools.yml]: no context here?

File: Integrations/integration-DomainTools.yml [hunk 6]
```diff
@@ -0,0 +1,541 @@
...
+                }
+            });
+        }
+
+        return {
+                Type: entryTypes.note,
+                Contents: res,
+                ContentsFormat: formats.json,
+                HumanReadable: 'Found '+numDomains +' domains:\n'+md,
+                EntryContext: context
```
Comment (Spivaka) [inline Integrations/integration-DomainTools.yml]: human readable format missing

File: Integrations/integration-DomainTools.yml [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+            address.domain_names.forEach(function(domain){
+                md += '* ' + domain + '\n';
+                context.Domain.push({'Name': domain, 'DNS' : {'Address' : address.ip_address}});
+            });
+        });
+
+        return {
+            Type: entryTypes.note,
+            Contents: res,
+            ContentsFormat: formats.json,
```
Comment (Spivaka) [inline Integrations/integration-DomainTools.yml]: again

File: Integrations/integration-DomainTools.yml (hunk 3 above)
Comment (shaniacht1) [inline Integrations/integration-DomainTools.yml]: Not exactly. it only makes an array if there's more than 1 item.

File: Integrations/integration-DomainTools.yml (hunk 1 above)
Comment (shaniacht1) [inline Integrations/integration-DomainTools.yml]: done

File: Integrations/integration-DomainTools.yml (hunk 5 above)
Comment (shaniacht1) [inline Integrations/integration-DomainTools.yml]: nope

File: Integrations/integration-DomainTools.yml (hunk 7 above)
Comment (shaniacht1) [inline Integrations/integration-DomainTools.yml]: fixed

**Pull request merged**